src/joint.jl

```diff
@@ -1,3 +1,34 @@
+struct Bounds{T}
+    lower::T
+    upper::T
+
+    function Bounds(lower::T1, upper::T2) where {T1, T2}
+        @assert lower <= upper
+        new{promote_type(T1, T2)}(lower, upper)
+    end
+    Bounds{T}() where {T} = new{T}(typemin(T), typemax(T))
+end
+
+upper(b::Bounds) = b.upper
+lower(b::Bounds) = b.lower
+Base.:(==)(b1::Bounds, b2::Bounds) = b1.lower == b2.lower && b1.upper == b2.upper
+Base.show(io::IO, b::Bounds) = print(io, "(", lower(b), ", ", upper(b), ")")
+Base.clamp(x, b::Bounds) = clamp(x, b.lower, b.upper)
+Base.intersect(b1::Bounds, b2::Bounds) = Bounds(max(b1.lower, b2.lower), min(b1.upper, b2.upper))
```

Comment:
Hmm, at this point, should we just be using `IntervalSets.ClosedInterval`s?

Reply:
Maybe? I'm hesitant to bring in another dependency just for this, though. I'd say wait and see if we actually find ourselves wanting more features than the ones I've implemented here.